artemis/modules/crtsh.py

```diff
@@ -27,48 +34,67 @@ class CrtshScanner(ArtemisBase):
     # the target. We lock crt.sh instead in the run() method.
     lock_target = False
 
-    def query_sql(self, domain: str) -> set[str]:
-        ct_domains: set[str] = set()
-        conn = connect("postgresql://guest@crt.sh:5432/certwatch")
-        conn.set_session(readonly=True, autocommit=True)
-        with conn.cursor() as cursor:
-            # Validate characters as we are putting the domain inside a SQL query
-            assert all([c in string.ascii_letters + "-" + string.digits + "." for c in domain])
-            cursor.execute(
-                (
-                    f"SELECT name_value FROM certificate_and_identities"
-                    f" WHERE plainto_tsquery('certwatch', '{domain}') @@ identities(certificate)"
-                    f" AND name_value ILIKE '%.{domain}'"
-                    f" AND (name_type = '2.5.4.3' OR name_type = 'san:dNSName')"
-                    f" GROUP BY name_value"
+    def query_sql(self, domain: str) -> Optional[Set[str]]:
+        try:
+            ct_domains: set[str] = set()
+            conn = connect("postgresql://guest@crt.sh:5432/certwatch")
+            conn.set_session(readonly=True, autocommit=True)
+            with conn.cursor() as cursor:
+                # Validate characters as we are putting the domain inside a LIKE query
+                # Escaping LIKE queries is complex:
+                # https://stackoverflow.com/questions/2106207/escape-sql-like-value-for-postgres-with-psycopg2
+                # so for now let's just whitelist domain characters.
+                assert all([c in string.ascii_letters + "-" + string.digits + "." or ord(c) >= 0x80 for c in domain])
+                cursor.execute(
+                    (
+                        "SELECT name_value FROM certificate_and_identities"
+                        " WHERE plainto_tsquery('certwatch', %s) @@ identities(certificate)"
+                        " AND name_value ILIKE %s"
+                        " AND (name_type = '2.5.4.3' OR name_type = 'san:dNSName')"
+                        " GROUP BY name_value"
+                    ),
+                    (domain, "%." + domain),
                 )
-            )
-            results = cursor.fetchall()
+                results = cursor.fetchall()
 
-            for row in results:
-                for entry in row:
-                    if re.fullmatch(DOMAIN_REGEX, entry):
-                        ct_domains.add(entry)
-        conn.close()
-        return ct_domains
+                for row in results:
+                    for entry in row:
```

Comment:
Why do we iterate the row if it returns a single column?

Reply:
fixed